### PR TITLE
Enable dashboard linter for pre-commit hooks

### PR DESCRIPTION
**Pull request by selfcontained**
## Description

This adds dashboard linting to pre-commit hooks so we won't commit any error/warnings the linter spots.

Linter errors can be seen in the git hook output or by manually running the linter.

```bash
# w/ leeway
leeway run components/dashboard:lint
# or from the dashboard component directory w/ yarn
yarn run lint
```

## Related Issue(s)

Fixes WEB-111

## How to test

I was able to test this while making these changes, but changing any ts/js file in the dashboard component so it has a linter error/warning, and trying to commit it should cause the hook to fail. 

#### Preview status

gitpod:summary

## Build Options:

- [ ] /werft with-werft
      Run the build with werft instead of GHA
- [ ] leeway-no-cache
- [ ] /werft no-test
      Run Leeway with `--dont-test`

<details>
<summary>Publish Options</summary>

- [ ] /werft publish-to-npm
- [ ] /werft publish-to-jb-marketplace
</details>

<details>
<summary>Installer Options</summary>

- [ ] analytics=segment
- [ ] with-dedicated-emulation
- [ ] with-ws-manager-mk2
- [ ] workspace-feature-flags
  Add desired feature flags to the end of the line above, space separated
</details>

#### Preview Environment Options:
- [ ] /werft with-local-preview
      If enabled this will build `install/preview`
- [ ] /werft with-preview
- [ ] /werft with-large-vm
- [ ] /werft with-gce-vm
      If enabled this will create the environment on GCE infra
- [ ] with-integration-tests=all
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`

/hold

### CODE REVIEW

**Review by svenefftinge — APPROVED** [review]

File: components/dashboard/package.json [hunk 1]
```diff
@@ -71,7 +71,7 @@
   "scripts": {
     "start": "craco start",
     "build": "craco build --verbose",
-    "lint": "eslint --ext=.jsx,.js,.tsx,.ts ./src",
+    "lint": "eslint --max-warnings=0 --ext=.jsx,.js,.tsx,.ts ./src",
```
Comment (easyCZ) [inline components/dashboard/package.json:74]: Can we change it such that it applies the linter to only modified files? I believe that should be done by passing the files into this command, and in the pre-commit enabling the `pass_filenames` feature. 

That way, we don't need to restrict it to just dashboard, but can enable for all ts/js files in incremental mode

Comment (geropl) [reply components/dashboard/package.json:74]: Discussed in [this loom](https://gitpod.slack.com/archives/C02EN94AEPL/p1682376030511889), but sounds like it's not straight forward.

But would be great to have!

Comment (geropl) [reply components/dashboard/package.json:74]: Maybe we already tried, but [this](https://gist.github.com/seeliang/0f0de424d1cdc4541c338f4ee93b7e6a#ci-script) looks promising.

Comment (easyCZ) [reply components/dashboard/package.json:74]: Let's land as is, and we can do a follow-up.

Comment (selfcontained) [reply components/dashboard/package.json:72]: I've tweaked things in the leeway command to use git diff to help filter out the files to run against and it seems to work as we want.

https://github.com/gitpod-io/gitpod/pull/17355/files#diff-b5cfb13a17f4f90cdd041c23e0b5ff56941b32cd8def8a743476e0cc0944b5cbR56

File: components/dashboard/src/menu/OrganizationSelector.tsx [hunk 2]
```diff
@@ -20,7 +19,6 @@ export default function OrganizationSelector() {
     const orgs = useOrganizations();
     const currentOrg = useCurrentOrg();
     const { data: userBillingMode } = useUserBillingMode();
-    const { data: orgBillingMode } = useOrgBillingMode();
```
Comment (selfcontained) [inline components/dashboard/src/menu/OrganizationSelector.tsx:23]: This is unused now, removing it.

File: components/dashboard/src/onboarding/StepOrgInfo.tsx [hunk 3]
```diff
@@ -15,7 +15,6 @@ import { EXPLORE_REASON_WORK, getExplorationReasons } from "./exploration-reason
 import { getJobRoleOptions, JOB_ROLE_OTHER } from "./job-roles";
 import { OnboardingStep } from "./OnboardingStep";
 import { getSignupGoalsOptions, SIGNUP_GOALS_OTHER } from "./signup-goals";
-import isURL from "validator/lib/isURL";
```
Comment (selfcontained) [inline components/dashboard/src/onboarding/StepOrgInfo.tsx:18]: unused, removing it.

File: components/dashboard/src/admin/UserDetail.tsx [hunk 4]
```diff
@@ -55,9 +55,6 @@ export default function UserDetail(p: { user: User }) {
         setUsageLimit(costCenter.spendingLimit);
     }, [costCenter, user]);
 
-    const email = User.getPrimaryEmail(p.user);
```
Comment (selfcontained) [inline components/dashboard/src/admin/UserDetail.tsx:58]: Both of these are unused

**Pull request merged**